module/modmanager/manager.go

```diff
@@ -121,6 +135,83 @@ func (mgr *Manager) Add(ctx context.Context, cfg config.Module) error {
 	return nil
 }
 
+// Reconfigure reconfigures an existing resource module and returns the names of
+// now orphaned resources.
+func (mgr *Manager) Reconfigure(ctx context.Context, cfg config.Module) ([]resource.Name, error) {
+	mod, exists := mgr.modules[cfg.Name]
+	if !exists {
+		return nil, errors.Errorf("cannot reconfigure module %s as it does not exist", cfg.Name)
+	}
+	handledResources := mod.resources
+
+	if err := mgr.remove(mod); err != nil {
+		return nil, err
+	}
+
+	if err := mgr.add(ctx, cfg); err != nil {
+		return nil, err
+	}
```

Comment:
These shouldn't be returning nil... if either of these fail, the orphaned resources are still actually orphaned and need to be closed and cleaned up.

Reply:
Updated here and in a few other places to return all the module's handled resources.